server/aws-lsp-partiql/src/server/language-service.ts

```diff
@@ -35,4 +60,19 @@ class PartiQLLanguageService {
         }
         return diagnostics
     }
+
+    public async doSemanticTokens(textDocument: TextDocument): Promise<SemanticTokens | null> {
+        const text = textDocument.getText()
+        const data: SemanticToken[] = []
+        for (const nodeType of semanticTokensLegend.tokenTypes) {
```

Comment:
Mainly just a question, why do we have these 2 different things to loop over (`semanticTokensLegend.tokenTypes` and `string2TokenTypes`). What are the differences between those two?

Reply:
Both `semanticTokensLegend.tokenTypes` and `string2TokenTypes` are node type defined in tree-sitter parser, where `semanticTokensLegend.tokenTypes` are the same type defined in LSP and `string2TokenTypes` are manually mapped to some type in LSP. 
For example, the `,` in `a, b` has type `operator` in tree-sitter and `,` in `{'a':1, 'b':1}` has type `tuple_punc_separator`. And the later one will be manually mapped to `SemanticTokenTypes.operator` defined in LSP.
I will combine the `semanticTokensLegend.tokenTypes` into `string2TokenTypes` if the code looks cleaner to you in that case. For now, I just keep them in two different lists to avoid overlapped definition and repeated code. Also, when adding a new `SemanticTokenTypes` support later, manual mapping will only be needed as a complementary.